api/app/src/main/kotlin/packit/service/UserRoleFilterService.kt

```diff
@@ -0,0 +1,153 @@
+package packit.service
+
+import org.springframework.stereotype.Service
+import packit.model.Packet
+import packit.model.Role
+import packit.model.User
+import packit.model.dto.RolesAndUsers
+import packit.security.PermissionChecker
+
+interface UserRoleFilterService
```

Comment:
The explosion of methods on here really makes me think that the `AccessKind` pattern described in my other comments would simplify things:

This is all that this service would really need to do:
```kt
fun getRoleAccessForPacketGroup(role: Role, name: String) -> List<AccessKind>
fun getRoleAccessForPacket(role: Role, packet: Packet) -> List<AccessKind>
```

- Role can be either a user or a non-username role
- The AccessKind is an enum describing the three possible states a role may have access to something.

Reply:
as per above comment